BlueprintUI/Tests/EnvironmentTests.swift

```diff
@@ -179,6 +179,188 @@ class EnvironmentTests: XCTestCase {
         enum RHSTestKey: EnvironmentKey { static let defaultValue: Int? = nil }
     }
 
+    func test_onDidRead() throws {
+
+        // Happy path
+        do {
+            var environment = Environment.empty
+
+            var readKey: Environment.StorageKey? = nil
+            let exp = expectation(description: "onDidRead")
+
+            environment.subscribeToReads(for: .layout) { key in
+                readKey = key
+                exp.fulfill()
+            }
+
+            environment[TestKey.self] = .right
+            _ = environment[TestKey.self]
+
+            wait(for: [exp], timeout: 0.1)
```

Comment:
Fwiw, this logging is synchronous, we shouldn't need this at all. Once `_ = environment[TestKey.self]` is performed, so are the tracking callbacks.

Reply:
- [ ] Remove test expectations